File: my_py_pkg/my_py_pkg/robot_listener.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
     
     
class ListenerNode(Node):
    def __init__(self):
        super().__init__("robot_listener") 
        self.subscriber_= self.create_subscription(String, "News", self.listener_callback, 10) # create subscriber
        self.get_logger().info("Listener started")

    def listener_callback(self, msg):
        self.get_logger().info(msg.data)
        #self.get_logger().info(f"Received: {msg.data}") shine
     
     
def main(args=None):
    rclpy.init(args=args)
    node = ListenerNode()
    rclpy.spin(node)
    rclpy.shutdown()
     
     
if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/robot_listener.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp" 
     
class RobotListenerNode : public rclcpp::Node // MODIFY NAME
{
public:
    // characteristics of the node
    RobotListenerNode() : Node("robot_listener") // MODIFY NAME
    {
        listener_ = this->create_subscription<example_interfaces::msg::String>(
            "News", 10, std::bind(&RobotListenerNode::listener_callback, this, std::placeholders::_1)); // MODIFY TOPIC NAME and add new placeholders for every argument on callback
        RCLCPP_INFO(this->get_logger(), "Robot Listener has been started");
    }
 
private:
    //private functions here
    void listener_callback(const example_interfaces::msg::String::SharedPtr msg) // MODIFY TOPIC NAME
    {
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
    }
    //declaring variables
    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr listener_; // MODIFY TOPIC NAME
};
 
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotListenerNode>(); // MODIFY NAME
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}